### DISCUSSION TITLE
Enable tcp socket in VICI and load configuration

### DISCUSSION THREAD

**Discussion by vishal-j** [strongswan/strongswan#1430] [Q&A]
I configured charon to use tcp socket for VICI. I am able to use swanctl. When using govici API, tcp connection is created. But I get connection reset by peer error while sending messages over it to run commands like getting version information. 
```
vici: transport error: read tcp 127.0.0.1:54030->127.0.0.1:4502: read: connection reset by peer
```
```
/etc/strongswan.d/charon/vici.conf
vici {

    # Whether to load the plugin. Can also be an integer to increase the
    # priority of this plugin.
    load = yes

    # Socket the vici plugin serves clients.
    # socket = unix://${piddir}/charon.vici
     socket = tcp://0.0.0.0:4502

}
/etc/strongswan.d/swanctl.conf
swanctl {

    # Plugins to load in swanctl.
    # load =

    # VICI socket to connect to by default.
    # socket = unix://${piddir}/charon.vici
     socket = tcp://0.0.0.0:4502

}
```
Is there anything else that I am missing in configuration? How do I troubleshoot this?

**Comment by tobiasbrunner** [comment]
> When using govici API, tcp connection is created. But I get connection reset by peer error while sending messages over it to run commands like getting version information.

I don't know much about the Go client. Perhaps you are doing that incorrectly. Is there anything logged by the plugin?

Maybe @enr0n (maintainer of [govici](https://github.com/strongswan/govici/)) has an idea.

**Comment by enr0n** [comment]
> `vici: transport error: read tcp 127.0.0.1:54030->127.0.0.1:4502: read: connection reset by peer`

This suggests to me that something happened on the daemon side, but it is hard to say without any logs.

@vishal-j In case it is on the client side, can you also share a go code snippet of what you are trying?

**Reply by vishal-j** [reply]
I was able to resolve it. There was an issue exposing port on the daemon side. Thanks for looking into it.
